java-squid/src/main/java/org/sonar/java/ast/AstScanner.java

```diff
@@ -94,15 +95,18 @@ public void simpleScan(Iterable<File> files) {
     progressReport.start(Lists.newArrayList(files));
     for (File file : files) {
       context.setFile(file);
+      context.addSourceCode(new SourceFile(file.getAbsolutePath(), file.getPath()));
```

Comment:
This should probably be added in the context.setFile method to get rid of the other two popSourceCode call (as setFile will pop code to project anyway)